vllm/v1/worker/gpu_model_runner.py

```diff
@@ -484,14 +484,11 @@ def _prepare_inputs(
         self.input_batch.block_table.commit(num_reqs)
 
         # Get the number of scheduled tokens for each request.
-        # TODO: The Python loop can be slow. Optimize.
-        num_scheduled_tokens = np.empty(num_reqs, dtype=np.int32)
-        max_num_scheduled_tokens = 0
-        for i, req_id in enumerate(self.input_batch.req_ids):
-            num_tokens = scheduler_output.num_scheduled_tokens[req_id]
-            num_scheduled_tokens[i] = num_tokens
-            max_num_scheduled_tokens = max(max_num_scheduled_tokens,
-                                           num_tokens)
+        req_ids = self.input_batch.req_ids
+        tokens = list(
+            map(scheduler_output.num_scheduled_tokens.__getitem__, req_ids))
```

Comment:
This may be a bit more readable?
```suggestion
        tokens = [scheduler_output.num_scheduled_tokens[i] for i in req_ids]
```

Reply:
Thanks for your suggestion! I have updated the code.